.taskcluster.yml

```diff
@@ -48,168 +44,210 @@ tasks:
+              taskId: '${ownTaskId}'
+              taskGroupId:
+                  $if: 'tasks_for == "action"'
+                  then:
+                      '${action.taskGroupId}'
+                  else:
+                      '${ownTaskId}'  # same as taskId; this is how automation identifies a decision task
+              schedulerId: '${trustDomain}-level-${level}'
+              created: {$fromNow: ''}
+              deadline: {$fromNow: '1 day'}
+              expires: {$fromNow: '1 year 1 second'}  # 1 second so artifacts expire first, despite rounding errors
+              metadata:
+                $merge:
+                  - owner: "${ownerEmail}"
+                    source: '${repoUrl}/raw/${head_sha}/.taskcluster.yml'
+                  - $if: 'tasks_for in ["github-push", "github-pull-request"]'
+                    then:
+                      name: "Decision Task"
+                      description: 'The task that creates all of the other tasks in the task graph'
+              provisionerId: "glean-${level}"
+              workerType: "decision"
+              tags:
+                kind: decision-task
+              routes:
+                - checks
+              scopes:
+                # `https://` is 8 characters so, ${repoUrl[8:]} is the repository without the protocol.
+                $if: 'tasks_for == "github-push"'
+                then:
+                  $let:
+                    short_head_branch:
+                      $if: 'head_branch[:11] == "refs/heads/"'
+                      then: {$eval: 'head_branch[11:]'}
+                      else: ${head_branch}
+                  in:
+                    - 'assume:repo:${repoUrl[8:]}:branch:${short_head_branch}'
+                else:
+                  $if: 'tasks_for == "github-pull-request"'
+                  then:
+                    - 'assume:repo:github.com/${event.pull_request.base.repo.full_name}:pull-request'
+
+              requires: all-completed
+              priority: lowest
+              retries: 5
+
+              payload:
+                env:
+                  # run-task uses these to check out the source; the inputs
+                  # to `mach taskgraph decision` are all on the command line.
+                  $merge:
+                    - GLEAN_BASE_REPOSITORY: '${baseRepoUrl}'
+                      GLEAN_HEAD_REPOSITORY: '${repoUrl}'
+                      GLEAN_HEAD_REF: '${head_branch}'
+                      GLEAN_HEAD_REV: '${head_sha}'
+                      GLEAN_REPOSITORY_TYPE: git
+                      TASKGRAPH_BASE_REPOSITORY: https://hg.mozilla.org/ci/taskgraph
+                      TASKGRAPH_HEAD_REPOSITORY: https://hg.mozilla.org/ci/${taskgraph.branch}
+                      TASKGRAPH_HEAD_REV: ${taskgraph.revision}
+                      TASKGRAPH_REPOSITORY_TYPE: hg
+                      REPOSITORIES: {$json: {appservices: "Application Services", taskgraph: "Taskgraph"}}
```

Comment:
Nit: s/APPservices/GLEAN here? so that we can use `--glean-checkout` below at line 208?